### ISSUE TITLE
Authentication error or Unsupported scheme when PBS baseURL uses HTTP and not HTTPS

### ISSUE THREAD

**Issue by nosoop4u** [tizbac/proxmoxbackupclient_go#13]
Note that when I connect to the PBS Web UI, it uses HTTP and not HTTPS - I think this might be the issue but I'm not sure.

This is on a Windows 10 22H2 client, connecting to PBS 3.0-2

The user for whom I created the access token is the same service account that is used on a couple of different PVE servers that run successful backups to this same PBS.

Just confirming, the values for the parameters passed are:
- "authid" is the token ID that was displayed when creating the API Token in the PBS UI under Configuration/Access Control
- "secret" is  the token secret from the same dialog as above
- "certfingerprint" is the fingerprint of the PBS proxy.pem cert as found under Configuration/Certificates

Tried to use a HTTP baseURL, with the command:
```
./proxmoxbackupgo_cli -authid "s_backup@pbs!win-pbs-cli" -backupdir 'f:\test' -baseurl "http://pbs1.<mydomain>:8007" -certfingerprint "<pbs-cert-fingerprint>" -datastore "nas1-bkp" -secret "<api-token-secret>"
```
and the result was:
```
Creating VSS Snapshot...Snapshot created: {7509F5CA-5FF2-4747-B8FB-3AE7C6947927}
Symlink from:  \\?\GLOBALROOT\Device\HarddiskVolumeShadowCopy7\  to:  C:\Users\<windows-username>\AppData\Roaming\PBSBackupGO\VSS\{7509F5CA-5FF2-4747-B8FB-3AE7C6947927}
Starting backup of C:\Users\<windows-username>\AppData\Roaming\PBSBackupGO\VSS\{7509F5CA-5FF2-4747-B8FB-3AE7C6947927}\test
Error making request: Get "http://pbs1.<mydomain>:8007/previous?archive-name=backup.pxar.didx": http2: unsupported scheme
panic: Get "http://pbs1.<mydomain>:8007/previous?archive-name=backup.pxar.didx": http2: unsupported scheme

goroutine 1 [running, locked to thread]:
main.(*PBSClient).DownloadPreviousToBytes(0xc000158b00, {0x7ff68dee43b2, 0x10})
        C:/Users/Utente/proxmoxbackupclient_go/pbsapi.go:428 +0x525
main.main()
        C:/Users/Utente/proxmoxbackupclient_go/main.go:113 +0x57e
```

Then, attempted to run the command using a HTTPS baseURL:
```
./proxmoxbackupgo_cli -authid "s_backup@pbs!win-pbs-cli" -backupdir 'f:\test' -baseurl "https://pbs1.<mydomain>:8007" -certfingerprint "<pbs-cert-fingerprint>" -datastore "nas1-bkp" -secret "<api-token-secret>"
```
and get the following:
```
Found leftover snapshot, deleting it...
Creating VSS Snapshot...Snapshot created: {81F811C9-E50F-4370-9FB2-2CEAA2A2FE4E}
Symlink from:  \\?\GLOBALROOT\Device\HarddiskVolumeShadowCopy6\  to:  C:\Users\<windows-username>\AppData\Roaming\PBSBackupGO\VSS\{81F811C9-E50F-4370-9FB2-2CEAA2A2FE4E}
Starting backup of C:\Users\<windows-username>\AppData\Roaming\PBSBackupGO\VSS\{81F811C9-E50F-4370-9FB2-2CEAA2A2FE4E}\test
Reading response to upgrade...

Unexpected response code: 401 Unauthorized
Error making request: Get "https://pbs1.<mydomain>:8007/previous?archive-name=backup.pxar.didx": Authentication error
panic: Get "https://pbs1.<mydomain>:8007/previous?archive-name=backup.pxar.didx": Authentication error

goroutine 1 [running, locked to thread]:
main.(*PBSClient).DownloadPreviousToBytes(0xc0000b6580, {0x7ff68dee43b2, 0x10})
        C:/Users/Utente/proxmoxbackupclient_go/pbsapi.go:428 +0x525
main.main()
        C:/Users/Utente/proxmoxbackupclient_go/main.go:113 +0x57e
```

**Comment by tizbac** [comment]
Hmm that may be a nasty issue because HTTP2 officially does not support plaintext at all, i have some memory of an extension maybe ending with C having support for it

**Comment by nosoop4u** [comment]
I may have misinterpreted what I was seeing in my browser.  It *is* using HTTPS, it's just reporting it as "not secure" because it's using the default self-signed certificate.  Here's what Chrome shows about the cert:

![pbs self-signed cert](https://github.com/tizbac/proxmoxbackupclient_go/assets/64984124/d219041d-1507-40ba-b5ef-50252885260f)

So, let's ignore the first attempt above where I used an HTTP baseURL and focus on the second where I used the proper HTTPS baseURL and got an authentication error.

For one thing, I noticed there that the cert name (and the CA name) does not match the DNS name of my PBS server - the cert is associated with ``pbs.<mydomain>`` but the server name is ``pbs1.<mydomain>``

Note this doesn't prevent my PVE servers from backing up to this PBS, as I used this same cert fingerprint when I added my PBS datastore to my PVE hosts.

I don't think this is the issue here, but I wanted to let you know just in case.  I just tried adding a DNS entry to my local DNS to make ``pbs.<mydomain>`` resolve to the PBS server IP, and tried the command again using ``pbs.<mydomain>`` in the baseURL and I got the same exact error (just with the different hostname).

So the question is, how do we troubleshoot this:
```
Unexpected response code: 401 Unauthorized
Error making request: Get "https://pbs.<mydomain>:8007/previous?archive-name=backup.pxar.didx": Authentication error
panic: Get "https://pbs.<mydomain>:8007/previous?archive-name=backup.pxar.didx": Authentication error

goroutine 1 [running, locked to thread]:
main.(*PBSClient).DownloadPreviousToBytes(0xc0000b6580, {0x7ff68dee43b2, 0x10})
        C:/Users/Utente/proxmoxbackupclient_go/pbsapi.go:428 +0x525
main.main()
        C:/Users/Utente/proxmoxbackupclient_go/main.go:113 +0x57e
```

Again, thanks so much for your help.

**Comment by tizbac** [comment]
Have you checked that also the token has proper permissions? command line looks correct 

Also token permissions are a subset of parent user permission

**Comment by nosoop4u** [comment]
I just came here to report I resolved the issue, by adding permissions to the API token!  I had assumed that permissions for the user associated with the token were sufficient, since I thought the token effectively connects as the user.  But yeah, great minds think alike...  The red highlight below indicates the newly added permission:
![API Token Permissions](https://github.com/tizbac/proxmoxbackupclient_go/assets/64984124/2dab9aaf-b811-49f6-8bb5-25067bccf1d2)

And now the backup works!  Thanks so much.
